src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -129,4 +129,68 @@ tnsr::ii<DataType, SpatialDim, Frame> extrinsic_curvature(
 template <size_t SpatialDim, typename Frame, typename DataType>
 tnsr::ijj<DataType, SpatialDim, Frame> deriv_spatial_metric(
     const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes spatial derivatives of lapse (N) from the generalized
+ *        harmonic variables and spacetime unit normal 1-form.
+ *
+ * \details If the generalized harmonic conjugate momentum and spatial
+ * derivative variables are \f$\Pi_{ab} = -t^c \partial_c \psi_{ab} \f$ and
+ * \f$\Phi_{iab} = \partial_i \psi_{ab} \f$, the spatial derivatives of N
+ * can be obtained from:
+ * \f{align*}
+ *  t^a t^b \Phi_{iab} = -1/(2N) [\partial_i (-N^2 + N_jN^j)-
+ *                               2 N^j \partial_i N_j
+ *                               + N^j N^k \partial_i g_{jk}]
+ *                     = -(2/N) \partial_i N,
+ * \f}
+ * since
+ * \f[
+ * \partial_i (N_jN^j) = 2N^j \partial_i N_j - N^j N^k \partial_i g_{jk}.
+ * \f]
+ *
+ * \f$ \Longrightarrow \partial_i N = -(N/2) t^a \Phi_{iab} t^b \f$
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::i<DataType, SpatialDim, Frame> spatial_deriv_of_lapse(
+    const Scalar<DataType>& lapse,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes time derivative of lapse (N) from the generalized
+ *        harmonic variables, lapse, shift and the spacetime unit normal 1-form.
+ *
+ * \details Let the generalized harmonic conjugate momentum and spatial
+ * derivative variables be \f$\Pi_{ab} = -t^c \partial_c \psi_{ab} \f$ and
+ * \f$\Phi_{iab} = \partial_i \psi_{ab} \f$, and the operator
+ * \f$D := \partial_0 - N^k \partial_k \f$. The time derivative of N is then:
+ * \f{align*}
+ *  (1/2) N^2 t^a t^b \Pi_{ab} - N N^i/2 t^a t^b \Phi_{iab}
+ *  =& (1/2) N^2 t^a t^b t^c \partial_c \psi_{ab}
+ *       - N N^i/2 (-(2/N) \partial_i N) \\
```

Comment:
I find `N N^i/2 (...)` confusing. Are you dividing by everything after the `/` or just the `2`? I think doing as @moxcodes suggested and using `\frac` (even if it's for now inconsistent with the rest of the file) would make the equations easier to parse :) I appreciate you actually showing the path to the result and not just the result!

Reply:
Agreed, changed to `\frac` .